src/main/java/InvalidIndexException.java

```diff
@@ -0,0 +1,2 @@
+public class InvalidIndexException extends InvalidInputException {
```

Comment:
Just a thought, are all the custom exceptions you define necessary? There might be some exceptions already defined in Java API that might be what you are looking for.